pkgs/development/cuda-modules/cuda-library-samples/generic.nix

```diff
@@ -125,22 +121,36 @@ in
           cuda_cccl # <nv/target>
         ];
 
-      postPatch = ''
-        substituteInPlace CMakeLists.txt \
-          --replace-fail "''${CUSPARSELT_ROOT}/lib64/libcusparseLt.so" "${lib.getLib cusparselt}/lib/libcusparseLt.so" \
-          --replace-fail "''${CUSPARSELT_ROOT}/lib64/libcusparseLt_static.a" "${lib.getStatic cusparselt}/lib/libcusparseLt_static.a"
-      '';
-
-      installPhase = ''
-        runHook preInstall
-        mkdir -p $out/bin
-        cp matmul_example $out/bin/
-        cp matmul_example_static $out/bin/
-        runHook postInstall
-      '';
+      postPatch =
+        prevAttrs.postPatch or ""
+        + ''
+          substituteInPlace CMakeLists.txt \
+            --replace-fail "''${CUSPARSELT_ROOT}/lib64/libcusparseLt.so" "${lib.getLib cusparselt}/lib/libcusparseLt.so" \
+            --replace-fail "''${CUSPARSELT_ROOT}/lib64/libcusparseLt_static.a" "${lib.getStatic cusparselt}/lib/libcusparseLt_static.a"
+        '';
+
+      installPhase =
+        prevAttrs.installPhase or ""
+        + ''
+          runHook preInstall
```

Comment:
Note:

- `preInstall` should probably run before `prevAttrs.installPhase`
- `prevAttrs.installPhase` might already be running the hooks

Reply:
I'd suggest you extend one of the (pre/post) hooks instead of the phase

Reply:
Switched to `postInstall` in the latest.